salon_creator/lib/screens/user_profile_edit_screen.dart

```diff
@@ -0,0 +1,320 @@
+
+        var userModel = UserModel.fromMap(snapshot.data.data());
+
+        return _buildUserProfileEditScreenInternal(userModel);
+      },
+    );
+  }
+
+  Future<bool> _onBackButtonPressed() async {
+    return !_profileUpdated
+        ? true
+        : await showDialog(
+              context: context,
+              builder: (context) => new AlertDialog(
+                title: new Text('変更内容を保存しますか？'),
+                content: new Text('変更内容を保存して、後で編集を続けられるようにしますか'),
+                actions: <Widget>[
+                  OutlinedButton(
+                    onPressed: () => Navigator.of(context).pop(false),
+                    child: Text('キャンセル'),
+                  ),
+                  ElevatedButton(
+                    onPressed: () => Navigator.of(context).pop(true),
+                    child: Text('削除'),
+                  ),
+                ],
+              ),
+            ) ??
+            false;
+  }
+
+  Widget _buildUserProfileEditScreenInternal(UserModel userModel) {
+    List<Widget> profileItems = [];
+    profileItems.addAll(ListTile.divideTiles(
+      context: context,
+      tiles: [
+        _buildProfileImageEditDisplay(userModel),
+        _buildNameEditDisplay(userModel),
+        _buildAboutMeEditDisplay(userModel),
+      ],
+    ).toList());
+
+    if (_opInProgress) {
+      profileItems.insert(0, LinearProgressIndicator());
+    }
+
+    return WillPopScope(
+      child: Scaffold(
+        appBar: _buildAppBar(userModel),
+        body: Padding(
+          padding: const EdgeInsets.symmetric(horizontal: 16),
+          child: ListView(
+            children: profileItems,
+          ),
+        ),
+      ),
+      onWillPop: _onBackButtonPressed,
+    );
+  }
+
+  Widget _buildAppBar(UserModel userModel) {
+    return AppBar(
+      title: Text('プロフィールの編集'),
+      iconTheme: IconThemeData(color: Colors.black),
+      actions: [
+        TextButton(
+          onPressed: _profileUpdated
+              ? () async {
+                  await _saveProfile(userModel);
+                  Navigator.pop(context);
+                }
+              : null,
+          child: Text('終了'),
+          style: ButtonStyle(
+            foregroundColor: MaterialStateProperty.resolveWith((states) =>
+                states.any((element) => element == MaterialState.disabled)
+                    ? Colors.grey
+                    : primaryColor),
+          ),
+        ),
+      ],
+    );
+  }
+
+  Future<void> _saveProfile(UserModel userModel) async {
+    setState(() {
+      _opInProgress = true;
+    });
+
+    userModel.profile.name = nameController.text;
+    userModel.profile.aboutMeText = aboutMeController.text;
+    if (_imageFile != null) {
+      userModel.profile.imageUrl = await _uploadImageAndGetUrl(_imageFile);
+    }
+
+    await FirebaseFirestore.instance
+        .collection(constants.DBCollection.users)
+        .doc(FirebaseAuth.instance.currentUser.email)
+        .update(userModel.toMap());
```

Comment:
We should move this to dbHandler file now or later if it is available ?
```suggestion
      await FirebaseFirestore.instance
        .collection(constants.DBCollection.users)
        .doc(FirebaseAuth.instance.currentUser.email)
        .update(userModel.toMap());
```